crates/polars-core/src/chunked_array/comparison/mod.rs

```diff
@@ -652,7 +652,10 @@ where
 {
     match (lhs.len(), rhs.len()) {
         (_, 1) => {
-            let right = rhs.chunks()[0]
+            let right = rhs
+                .downcast_iter()
+                .find(|x| !x.is_empty())
```

Comment:
empty chunks 🥲

Reply:
The filter goes through a the `binary_unchecked_same_type` kernel, which takes the filter `op` as `Fn(Array, Array) -> Array` - so if the predicate was false for an entire chunk it ends up empty

https://github.com/pola-rs/polars/blob/8c39bb9f4fd1dd0baba596a60fbdaa7c109432fb/crates/polars-core/src/chunked_array/ops/arity.rs#L535-L541